src/backend/access/aocs/aocsam_handler.c

```diff
@@ -229,10 +229,15 @@ find_dml_state(const Oid relationOid)
 static inline void
 remove_dml_state(const Oid relationOid)
 {
+#ifdef USE_ASSERT_CHECKING
 	AOCODMLState *state;
+#endif
 	Assert(aocoDMLStates.state_table);
 
-	state = (AOCODMLState *) hash_search(aocoDMLStates.state_table,
+#ifdef USE_ASSERT_CHECKING
+	state = (AOCODMLState *)
+#endif
+		hash_search(aocoDMLStates.state_table,
```

Comment:
I suggest to use one ifdef and don't cast, because the type of `state` does not matter
```c
	Assert(aocoDMLStates.state_table);
#ifdef USE_ASSERT_CHECKING
	void* state =
#endif
		hash_search(aocoDMLStates.state_table,
```